gradle/tasks/generateListeners.gradle.kts

```diff
@@ -0,0 +1,261 @@
+            addInlineEventsToEventFile(eventFile, guildModerationListeners)
+
+            logger.lifecycle("Writing files")
+            File("${project.buildDir}/generated/kotlin").apply {
+                try {
+                    mkdirs()
+                    coreListenerFile.build().writeTo(this)
+                    guildListenerFile.build().writeTo(this)
+                    channelListenerFile.build().writeTo(this)
+                    userListenerFile.build().writeTo(this)
+                    voiceListenerFile.build().writeTo(this)
+                    interactionListenerFile.build().writeTo(this)
+                    guildModerationListenerFile.build().writeTo(this)
+
+                    eventFile.build().writeTo(this)
+
+                    logger.lifecycle("Done")
+                } catch (e: Exception) {
+                    e.printStackTrace()
+                    logger.error("Failed to write files")
+                }
+            }
+        }
+    } catch (e: Exception) {
+        e.printStackTrace()
+        logger.error("Failed to generate files")
+    }
+}
+
+fun addFunctions(file: FileSpec.Builder, listeners: List<ClassInfo>) {
+    listeners.forEach { listener ->
+        // get the name of the class and add "on" to the front
+        val name = listener.simpleName
+        // the paramater in the function is just the event class name but with a lowercase first
+        // letter
+        val eventParam =
+            listener.simpleName.first().toLowerCase() + listener.simpleName.substring(1)
+        val eventName = "on" + name
+        val eventPackage = listener.packageName
+
+        // add the function to the interface
+        file.addFunction(
+            FunSpec.builder(eventName)
+                .addParameter(eventParam, ClassName(eventPackage, name))
+                .build())
+    }
+}
+
+fun addOverideOnEvent(file: FileSpec.Builder, listeners: List<ClassInfo>) {
+    listeners.forEach { listener ->
+        // get the name of the class and add "on" to the front
+        val name = listener.simpleName
+        // the paramater in the function is just the event class name but with a lowercase first
+        // letter
+        val eventParam =
+            listener.simpleName.first().toLowerCase() + listener.simpleName.substring(1)
+        val eventName = "on" + name
+        val eventPackage = listener.packageName
+
+        file.addFunction(
+            FunSpec.builder("onEvent")
+                .addParameter(
+                    "event", ClassName("io.github.ydwk.ydwk.evm.backend.event.", "GenericEvent"))
+                .addCode(
+                    CodeBlock.builder()
+                        .beginControlFlow("when (event)")
+                        .addStatement(
+                            "is %T -> %L(event)", ClassName(eventPackage, name), eventName)
+                        .endControlFlow()
+                        .build())
+                .build())
+    }
+}
```

Comment:
``` Kotlin

val restAPIMethodGettersInterface =
        FileSpec.builder("io.github.ydwk.yde.rest.methods", "RestAPIMethodGetters")
            .addType(
                TypeSpec.interfaceBuilder("RestAPIMethodGetters")
                    .addModifiers(KModifier.PUBLIC)
                    .addAnnotation(
                        AnnotationSpec.builder(Suppress::class).addMember("\"UNUSED\"").build())
                    .addAnnotation(
                        AnnotationSpec.builder(
                                ClassName("javax.annotation.processing", "Generated"))
                            .addMember("\"Generated by YDE\"")
                            .addMember("date = " + "\"$formattedDateAndTime\"")
                            .build())
                    .addFunctions(
                        interfaceNemes.map {
                            FunSpec.builder("get${it}")
                                .addModifiers(KModifier.ABSTRACT)
                                .returns(ClassName("io.github.ydwk.yde.rest.methods", it))
                                .build()
                        })
                    .build())
            .build()
```


Do something like this instead of having fucntions.